src/chocolatey/infrastructure.app/nuget/NugetList.cs

```diff
@@ -41,9 +41,9 @@ public static IEnumerable<IPackageSearchMetadata> GetPackages(ChocolateyConfigur
             return SearchPackagesAsync(configuration, nugetLogger, filesystem).GetAwaiter().GetResult();
         }
 
-        public static int GetCount(ChocolateyConfiguration configuration, ILogger nugetLogger, IFileSystem filesystem)
+        public static int GetCount(ChocolateyConfiguration configuration, ILogger nugetLogger, IFileSystem filesystem, ChocolateySourceCacheContext cacheContext)

```

Comment:
We need to add this as an overload, instead of changing the existing one (backwards compatibility).

```suggestion
        public static int GetCount(ChocolateyConfiguration configuration, ILoger nugetLogger, IFileSystem fileSystem)
        {
            return GetCount(configuration, nugetLogger, fileSystem, new ChocolateySourceCacheContext());
        }

        public static int GetCount(ChocolateyConfiguration configuration, ILogger nugetLogger, IFileSystem filesystem, ChocolateySourceCacheContext cacheContext)
```

It should make the old method obsolete, though.

Reply:
Doh! Thanks for catching this!